src/meta/src/barrier/mod.rs

```diff
@@ -1195,3 +1234,15 @@ fn collect_synced_ssts(
     }
     (sst_to_worker, synced_ssts)
 }
+
+fn merge_compute_node_rpc_error(
+    message: &str,
+    errors: impl IntoIterator<Item = (WorkerId, RpcError)>,
+) -> MetaError {
+    use std::fmt::Write;
+    let concat: String = errors.into_iter().fold(String::new(), |mut s, (w, e)| {
+        write!(&mut s, " worker {}, {};", w, e).unwrap();
```

Comment:
This may lose the source chain. Try `.as_report()`: https://docs.rs/thiserror-ext/latest/thiserror_ext/trait.AsReport.html